pyrit/prompt_converter/leetspeak_converter.py

```diff
@@ -27,14 +27,14 @@ def __init__(self) -> None:
     def convert(self, *, prompt: str, input_type: PromptDataType = "text") -> ConverterResult:
         """
         Simple converter to generate leatspeak version of a prompt.
-        Since there are multiple character variations, this is non-deterministic
+        Since there are multiple character variations, this is non-deterministic.
         """
         if not self.input_supported(input_type):
             raise ValueError("Input type not supported")
 
         converted_prompt = []
-        for char in prompt.lower():
-            if char in self.leet_substitutions:
+        for char in prompt:
+            if char.lower() in self.leet_substitutions:
                 converted_prompt.append(random.choice(self.leet_substitutions[char]))
```

Comment:
I think we need it here, too.
```suggestion
                converted_prompt.append(random.choice(self.leet_substitutions[char.lower()]))
```

Reply:
Or perhaps `self.leet_substitutions.get(char.lower(), char)` is the way to go? Seems a bit simpler.